### PR TITLE
rebase all docker images on alpine linux

### PR DESCRIPTION
**Pull request by cocoon-project**
much smaller

### CODE REVIEW

File: api/app.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+Executable for the Ceryx server.
+"""
+
+from ceryx import settings
+from ceryx.api import app
+
+if __name__ == '__main__':
+
+    app.run(host=settings.API_BIND_HOST, port=settings.API_BIND_PORT, debug=True)
+    #app.run( host="0.0.0.0", port=80, debug=True )
```
Comment (akalipetis) [inline api/app.py]: Could you please remove this comment?

File: api/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+redis==2.10.3
```
Comment (akalipetis) [inline api/requirements.txt]: Since this introduces a `requirements.txt` file, I believe it would be best if you sourced these requirements to the `setup.py` file too, in order to keep requirements in a centralized location.

ie, you can change [this line](https://github.com/sourcelair/ceryx/blob/master/api/setup.py#L15) to something like:

```python
install_requires=requirements.read().split('\n')
```

File: docker-compose.yml [hunk 3]
```diff
@@ -7,19 +7,19 @@ proxy:
...
     file: base.yml
     service: api
-  build: api
+  image: jazzdd/alpine-flask
   net: container:proxy
   volumes:
-   - api/ceryx:/ceryx/ceryx
-   - api/bin:/ceryx/bin
+   - ./api:/app
+  command: "-d"
```
Comment (akalipetis) [inline docker-compose.yml]: From a quick testing, I could not make this work. Does this miss any arguments?

File: docker-compose.yml [hunk 4]
```diff
@@ -7,19 +7,19 @@ proxy:
...
+   - ./proxy/nginx/conf:/opt/openresty/nginx/conf
+   - ./proxy/nginx/lualib:/opt/openresty/nginx/lualib
+   - ./proxy/nginx/sites-enabled:/opt/openresty/nginx/sites-enabled
+   - ./proxy/nginx/sites-available:/opt/openresty/nginx/sites-available
 api:
   extends:
     file: base.yml
     service: api
-  build: api
+  image: jazzdd/alpine-flask
```
Comment (akalipetis) [inline docker-compose.yml]: Shouldn't we use `python:alpine` as base and build the requirements in? Otherwise, the requirements are missing and this `docker-compose` file will fail I think.

**Comment by akalipetis** [comment]
Thanks a lot @cocoon-project for this contribution :cookie: 

I've made some comments, but I think they are trivial and we'll soon get this merged. Really good idea to lower the image sizes here.

File: docker-compose.yml (hunk 3 above)
Comment (cocoon-project) [inline docker-compose.yml]: no argument is missing, there is a entry_point in the docker file,
the argument is -d ( debug mode) because we dont use the nginx mode of the image

File: api/requirements.txt (hunk 2 above)
Comment (cocoon-project) [inline api/requirements.txt]: The explicit requirements.txt is needed, see the doc of the docker image, there is an auto build

File: docker-compose.yml (hunk 4 above)
Comment (cocoon-project) [inline docker-compose.yml]: I will have look to the python-alpine image.

File: api/requirements.txt (hunk 2 above)
Comment (akalipetis) [inline api/requirements.txt]: It sure does, it would just be optimal that it was in sync with `setup.py` - see reference code above.

File: docker-compose.yml (hunk 3 above)
Comment (akalipetis) [inline docker-compose.yml]: Ok, just checked. I don't believe it would be a good idea to have two processes that are not related in the same container. We should either keep the existing image here, or use the official Alpine flavor [`python2.7.11-alpine`](https://hub.docker.com/_/python/)

**Comment by akalipetis** [comment]
@cocoon-project I'm closing this due to inactivity. If you find the time to get on top of it again, feel free to reopen it.
